derive/src/impl_wrapper.rs

```diff
@@ -12,32 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#[cfg(not(feature = "std"))]
-use alloc::{
-    format,
-    string::ToString,
-};
-
 use proc_macro2::{
     Span,
     TokenStream as TokenStream2,
 };
 use quote::quote;
 use syn::Ident;
 
-pub fn wrap(
-    ident: &Ident,
-    trait_name: &'static str,
-    impl_quote: TokenStream2,
-) -> TokenStream2 {
-    let mut renamed = format!("_IMPL_{}_FOR_", trait_name);
-    renamed.push_str(ident.to_string().trim_start_matches("r#"));
-    let dummy_const = Ident::new(&renamed, Span::call_site());
+pub fn wrap(impl_quote: TokenStream2) -> TokenStream2 {
+    let include_scale_info = include_crate("scale-info", "_scale_info");
+    let include_parity_scale_codec = include_crate("parity-scale-codec", "_scale");
 
     quote! {
         #[allow(non_upper_case_globals, unused_attributes, unused_qualifications)]
-        const #dummy_const: () = {
+        const _: () = {
+            #include_scale_info
+            #include_parity_scale_codec
+
             #impl_quote;
         };
     }
 }
+
+/// Include a crate under a known alias, to be robust against renamed dependencies.
+fn include_crate(name: &str, alias: &str) -> proc_macro2::TokenStream {
+    match proc_macro_crate::crate_name(name) {
+        Ok(crate_name) => {
+            let crate_name_ident = Ident::new(&crate_name, Span::call_site());
+            let crate_alias_ident = Ident::new(&alias, Span::call_site());
+            quote!( extern crate #crate_name_ident as #crate_alias_ident; )
+        }
+        Err(e) => syn::Error::new(Span::call_site(), &e).to_compile_error(),
```

Comment:
I'd personally prefer a solution where we'd still keep having absolute paths everything.
This means that `include_crate` needed to return a `TokenStream` in the form of `quote! { codec }` or `quote { scale }` or whatever is the actual crate alias for it and users would then use it as:
```rust
let parity_scale_codec_alias = include_crate("parity-scale-codec");
quote! { :: #parity_scale_codec_alias ::rest::of::my::path }
```
This solution would also not need yet another alias, e.g. `_scale_info`.